### ISSUE TITLE
Using setSchema in Builder overwrites scim core schema

### ISSUE THREAD

**Issue by tkrille** [osiam/scim-schema#127]
Moved from [Jira](https://jira.osiam.org/browse/BT-9):

> When using Resource.Builder.setSchema() the scim core schema gets overwritten unless you specify it explicitly in the parameter Set

**Comment by tkrille** [comment]
We should take care that the core schema is always present.

**Comment by tkrille** [comment]
Ok, I looked at this. For me it seems, that setting the schemata is totally senseless. If no extension are involved, then adding another schema or even overwrite the core schemata, would mean you have a different resource, i.e. not a `User` but an `EnterpriseUser`, which is not possible, without adding new fields, etc. to the existing classes. If extensions are involved, then adding the extension will automatically add the URN to the list of schemata.

Long story short: I will deprecate the methods, including a hint, that the method will be finally removed in version 1.8 or 2.0. 3 versions should be enough for users to update. Hopefully, nobody is using the method anyway.
